test/e2e/scalenodes.go

```diff
@@ -75,11 +81,41 @@ var _ = Describe("Scale nodes", func() {
 		err = waitForScale(mss.Items[0].Name)
 		Expect(err).NotTo(HaveOccurred())
 	})
+
+	Specify("operator should maintain at least three worker replicas", func() {
+		ctx := context.Background()
+		infraId, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		mss, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).List(ctx, metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(mss.Items).NotTo(BeEmpty())
+
+		// Remove a replica from a worker MachineSet, and wait
+		err = scale(mss.Items[0].Name, -1)
+		Expect(err).NotTo(HaveOccurred())
+
+		ms, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).Get(ctx, mss.Items[0].Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		err = waitForScale(mss.Items[0].Name)
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
There is a potential for a flake:

1. We scale
2. We read machine set
3. We wait for scale to complete
4. We do aserts based on the object from step 2

Operator will try to scale workers back after step 1. If this happenes before we read on step 2 - we are good (I guess that's why currently e2es are green) If it happens after step 2 - we will see a failure.

Reply:
I've added the wait before the object is read rather than after to hopefully avoid the race.